data-prepper-core/src/main/java/org/opensearch/dataprepper/core/pipeline/buffer/ZeroBuffer.java

```diff
@@ -76,11 +79,18 @@ public void writeAll(Collection<T> records, int timeoutInMillis) throws Exceptio
     @Override
     public Map.Entry<Collection<T>, CheckpointState> read(int timeoutInMillis) {
         if (threadLocalStore.get() == null) {
-            threadLocalStore.set(new ArrayList<>());
+            try {
+                Thread.sleep(DEFAULT_READ_SLEEP_MILLIS);
+            } catch (InterruptedException e) {
+                // Restore the interrupted status
+                Thread.currentThread().interrupt();
```

Comment:
I do not think `Thread.currentThread().interrupt();` is needed. We do not need to restore the "interrupted" status.